pkg/cli/admin/inspect/inspect.go

```diff
@@ -117,6 +118,7 @@ func NewCmdInspect(streams genericiooptions.IOStreams) *cobra.Command {
 	cmd.Flags().BoolVarP(&o.allNamespaces, "all-namespaces", "A", o.allNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
 	cmd.Flags().StringVar(&o.sinceTime, "since-time", o.sinceTime, "Only return logs after a specific date (RFC3339). Defaults to all logs. Only one of since-time / since may be used.")
 	cmd.Flags().DurationVar(&o.since, "since", o.since, "Only return logs newer than a relative duration like 5s, 2m, or 3h. Defaults to all logs. Only one of since-time / since may be used.")
+	cmd.Flags().StringVar(&o.nodeSelector, "node-selector", o.nodeSelector, "Experimental: Set a specific node selector to use - only relevant for the daemonset pods running on data plane nodes, since the default may discard them from inspection.")
```

Comment:
As a user I'd expect to collect every pod when `--node-selector` is not specified. What about to set the node selector to `node-role.kubernetes.io/control-plane` by default? Presumably any node with `node-role.kubernetes.io/control-plane` is also labeled with `node-role.kubernetes.io/master` in our installations. So only the former is needed.

Also, this new experimental flag already changes the default behavior. So it can not be marked as experimental. Let's promote the flag to a GA flag and mark this as a bug fix as collecting the DS pods from data nodes has never been an intention to perform by default.

Reply:
Thank you for dropping your comment. I've updated the PR with respect to your changes. Could you PTAL?.